opticdev/optic
# Add update notifier to openapi-cli

## 🍗 Description
_What does this PR do? Anything folks should know?_

This adds `update-notifier` to the OAS cli, which is probably our best mechanism to encourage upgrading of the CLI to the latest version. We want to encourage this as the CLI is new and still a moving target. Checks and notifications happen at most once per day. 

![Screenshot 2022-08-04 at 12 49 13](https://user-images.githubusercontent.com/857549/182833450-25effdef-3ba1-4e30-b1aa-9ea81891def9.png)

When installed through npm, fully automated updates are not welcome, but notifying a user seems widely used and accepted.

It's been implemented in a way that allows the CLI to be hosted in another package. When passing it specific package information, the check and update instructions will happen for this 

## 👹 QA
_How can other humans verify that this PR is correct?_
- install prerelease `0.26.29-test.0` and run it twice. First time the check will happen and results persisted (in a separate process), the second will show the notification to update to `0.26.29-test.1`.
